### PR TITLE
Add libsql+file:// support

### LINKED ISSUE #2198: Can't connect to LibSQL local file database
I'm trying to use Atlas but if I set up my atlas.hcl as such:
```
env "local" {
  src = "file://schema.sql"
  url = "libsql://data/local.db"
  dev = "libsql://data/atlas.db"
}
```

Then I get this error (when for example I run a command like `atlas migrate diff --env local`):
```
Error: sqlite: query version pragma: failed to WebSocket dial: failed to send handshake request: Get "https://data/atlas.db": dial tcp: lookup data on 172.27.240.1:53: no such host
```

Does atlas only support LibSQL when run with Turso or as a server?

I could modify it to just use sqlite instead of libsql for local tests but then libsql supports some features which libsql does not (for example random row id), and so atlas would give an error when trying to use one of those

### PR DESCRIPTION
**Pull request by haaawk**
Fixes #2198

### CODE REVIEW

File: sql/sqlite/driver.go [hunk 1]
```diff
@@ -65,9 +65,13 @@ func init() {
...
 		sqlclient.RegisterTxOpener(OpenTx),
 		sqlclient.RegisterCodec(MarshalHCL, EvalHCL),
-		sqlclient.RegisterFlavours("libsql+ws", "libsql+wss"),
+		sqlclient.RegisterFlavours("libsql+ws", "libsql+wss", "libsql+file"),
 		sqlclient.RegisterURLParser(sqlclient.URLParserFunc(func(u *url.URL) *sqlclient.URL {
-			return &sqlclient.URL{URL: u, DSN: strings.TrimPrefix(u.String(), "libsql+"), Schema: mainFile}
+			dns := strings.TrimPrefix(u.String(), "libsql+")
+			if strings.HasPrefix(dns, "file://") {
+				dns = strings.Replace(dns, "file://", "file:", 1)
+			}
```
Comment (a8m) [inline sql/sqlite/driver.go:73]:
```suggestion
			dsn := strings.TrimPrefix(u.String(), "libsql+")
			if strings.HasPrefix(dns, "file://") {
				dsn = strings.Replace(dsn, "file://", "file:", 1)
			}
```

File: sql/sqlite/driver.go [hunk 2]
```diff
@@ -65,9 +65,13 @@ func init() {
...
+			return &sqlclient.URL{URL: u, DSN: dns, Schema: mainFile}
```
Comment (a8m) [inline sql/sqlite/driver.go:74]:
```suggestion
			return &sqlclient.URL{URL: u, DSN: dsn, Schema: mainFile}
```

**Comment by a8m** [comment]
CI is failing due to GitHub Actions issues. I'll rerun this soon

File: sql/sqlite/driver.go [hunk 3]
```diff
@@ -65,9 +65,13 @@ func init() {
 		sqlclient.DriverOpener(Open),
 		sqlclient.RegisterTxOpener(OpenTx),
 		sqlclient.RegisterCodec(MarshalHCL, EvalHCL),
-		sqlclient.RegisterFlavours("libsql+ws", "libsql+wss"),
+		sqlclient.RegisterFlavours("libsql+ws", "libsql+wss", "libsql+file"),
 		sqlclient.RegisterURLParser(sqlclient.URLParserFunc(func(u *url.URL) *sqlclient.URL {
-			return &sqlclient.URL{URL: u, DSN: strings.TrimPrefix(u.String(), "libsql+"), Schema: mainFile}
+			dsn := strings.TrimPrefix(u.String(), "libsql+")
+			if strings.HasPrefix(dns, "file://") {
```
Comment (giautm) [inline sql/sqlite/driver.go:71]:
```suggestion
			if strings.HasPrefix(dsn, "file://") {
```

**Comment by haaawk** [comment]
Thanks @giautm

**Review by giautm — APPROVED** [review]

**Comment by giautm** [comment]
Thank you for your contribution @haaawk,

**Pull request merged**